src/Middleware/ResponseCompression/src/BodyWrapperStream.cs

```diff
@@ -39,13 +39,9 @@ internal class BodyWrapperStream : Stream, IHttpBufferingFeature, IHttpSendFileF
             _innerSendFileFeature = innerSendFileFeature;
         }
 
-        protected override void Dispose(bool disposing)
+        internal ValueTask FinishCompressionAsync()
```

Comment:
We should probably add a regression test verifying that disposing the ResponseCompression stream doesn't end/poison the stream anymore.